zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js

```diff
@@ -759,6 +759,9 @@ function ParagraphCtrl($scope, $rootScope, $route, $window, $routeParams, $locat
   };
 
   $scope.sendPatch = function() {
+    if (!$scope.userHasWritePermission()) {
```

Comment:
should this save $scope.userHasWritePermission() and not called it multiple times?
what if the return from `$scope.userHasWritePermission()` changes?

Reply:
But permissions can change while the notebook is open. Or do you propose to calculate userHasWritePermission() only when the permissions change?